### ISSUE TITLE
Need a more natural window shadows

### ISSUE THREAD

**Issue by matinlotfali** [matinlotfali/KDE-Rounded-Corners#101]
From the Reddit comment [here](https://www.reddit.com/r/kde/comments/15ebhil/comment/jubu7ml/?utm_source=share&utm_medium=web2x&context=3):

> Your project is great, but I wish it provided more natural window shadows, not just dark colour glowing. Glowing-like shadows are no-go for me right now, unfortunately.

And my response:

> I agree that my shadows are ugly. I didn’t spend a lot of time to implement the math formula. I am open to change its math formula if I get some help.

**Comment by RomanKode** [comment]
My suggestion would be to look at how a-parhom/Lightlyshaders implemented his shadows and possibly use that as a foundation. for now, you can rebrand the current "shadow" that you have as a "glow" since it has that even gradient,

**Comment by matinlotfali** [comment]
LightlyShaders is grabbing the original plasma shadow and it is curving its corners. This is a resource-heavy computation which I am trying my best to avoid. 
I prefer to redraw shadows with mathematical equations instead.

**Comment by paletteOvO** [comment]
I also found the shadows are ugly. So I tried some formulas from [here](https://stackoverflow.com/questions/13462001/ease-in-and-ease-out-animation-formula). They definitely bring better results. I personally prefer ParametricBlend but they still don't look very "natural". Also, I think the shadows disappear too soon but I am unsure how to adjust.

shadowSize=20
Original:
![image](https://github.com/matinlotfali/KDE-Rounded-Corners/assets/16734391/862b1f00-a92a-465e-aaa9-dbd50738e44b)
BezierBlend:
![image](https://github.com/matinlotfali/KDE-Rounded-Corners/assets/16734391/a0890d93-88a7-4b3c-9294-ed1081f94b49)
ParametricBlend:
![image](https://github.com/matinlotfali/KDE-Rounded-Corners/assets/16734391/8f351dca-3856-4d12-b237-67c846f5b89a)

shadowSize=32
Original:
![image](https://github.com/matinlotfali/KDE-Rounded-Corners/assets/16734391/3c1a44f3-5161-40c5-9603-26dfec5ce5a4)
BezierBlend:
![image](https://github.com/matinlotfali/KDE-Rounded-Corners/assets/16734391/b005fb88-a9e9-40db-a072-eb466be72cf3)
ParametricBlend:
![image](https://github.com/matinlotfali/KDE-Rounded-Corners/assets/16734391/da6757cc-08ef-4457-ba08-e0b4aa48bbb1)

**Comment by matinlotfali** [comment]
They are all very beautiful and definitely better than the original (i.e Linear).
I also like the ParametricBlend better.
Can you open a pull request and add your name to README.md contributions?

**Comment by paletteOvO** [comment]
I am still trying different equations. But some results look strange to me. So I also rewrote some of the shaders. Would you like to take a look at it? 
https://github.com/paletteOvO/KDE-Rounded-Corners

Before:
shadowSize = 20, 100% opacity
![image](https://github.com/matinlotfali/KDE-Rounded-Corners/assets/16734391/563de515-acc4-447c-a8c9-3bb91f9e7e46)

After:
shadowSize = 4, 50% opacity
![image](https://github.com/matinlotfali/KDE-Rounded-Corners/assets/16734391/db3b7f8f-4dd9-4a31-a78a-134557d92cc6)

I also include another equation that may produce a more natural result:
reverse square, shadowSize = 20, 100% opacity
![image](https://github.com/matinlotfali/KDE-Rounded-Corners/assets/16734391/b55b9316-e639-4d40-ac08-17699ab31d50)

**Comment by matinlotfali** [comment]
Sure. I haven't written any comments in the shader. This might have caused a lot of headaches for you to figure out what has been happening. 
I can see you have already figured out the ShapeCorners and ShadowCorners functions. Excellent!
But for the main function, I can add comments so it would help you with the logic behind its variables. How is that?

**Comment by paletteOvO** [comment]
It would be helpful. I don't really understand the original distance_from_center and percent. And they don't seem fit to what I expected the percent should be from 1 to 0 count from the closest edge, which is the behavior of my version. You can see they produce different results in the image.

**Comment by RomanKode** [comment]
If it can be possible for options of customization, Bezier Blend and Parametric Blend should be included, this with color options and different size, opacity and (probably) offsets it can make great shadows to window glow(a'la Oxygen theme).

Also try Shadow Size 20, Opacity 80-90% I think it might look more natural as shadows, also adding some offset to the shadow should give shadow perspective.

**Comment by paletteOvO** [comment]
There are some possible customizations for invSquare and parametric easeInOut blend. However, I doubt allowing customizing of easeInOut enables the user to create a better effect, and it will need some extra calculation. But having different choices of the equation would be great.

a=2 and a=4 in parametric easeInOut blend:
![image](https://github.com/matinlotfali/KDE-Rounded-Corners/assets/16734391/7e16c73b-fd14-40e7-861b-a1d3ab2182ac)

a=1 and a=8 in inverse square blend
![image](https://github.com/matinlotfali/KDE-Rounded-Corners/assets/16734391/2af65410-3ed8-486d-b781-17dadbd95003)

**Comment by matinlotfali** [comment]
@paletteOvO I added a lot of comments for you in #121

**Comment by matinlotfali** [comment]
@paletteOvO Do I have your permission to include your edits in your fork to my code? 
Of course, I will mention your name in the `README.md`

**Comment by paletteOvO** [comment]
> @paletteOvO Do I have your permission to include your edits in your fork to my code? Of course, I will mention your name in the `README.md`

 Yes, of course. I am a bit busy right now so might not have time to create a pull request.

**Comment by matinlotfali** [comment]
May I have everyone test #125 and provide comments?
